optuna/integration/_lightgbm_tuner/sklearn.py

```diff
@@ -1,49 +1,887 @@
+                elif isinstance(collection, dict):
+                    return collection.get(i, None)
+                else:
+                    raise TypeError(
+                        "eval_sample_weight, eval_class_weight, eval_init_score, and eval_group "
+                        "should be dict or list"
+                    )
+
+            for i, (X_valid, y_valid) in enumerate(eval_set):
+                valid_class_weight = _get_meta_data(eval_class_weight, i)
+                valid_group = _get_meta_data(eval_group, i)
+                valid_init_score = _get_meta_data(eval_init_score, i)
+                valid_weight = _get_meta_data(eval_sample_weight, i)
+
+                if valid_class_weight is not None:
+                    valid_class_sample_weight = compute_sample_weight(valid_class_weight, y_valid)
+
+                    if valid_weight is None or len(valid_weight) == 0:
+                        valid_weight = valid_class_sample_weight
+                    else:
+                        valid_weight *= valid_class_sample_weight
+
+                valid_set = lgb.Dataset(
+                    X_valid,
+                    label=y_valid,
+                    weight=valid_weight,
+                    group=valid_group,
+                    init_score=valid_init_score,
+                    feature_name=feature_name,
+                    categorical_feature=categorical_feature,
+                )
+
+            valid_sets.append(valid_set)
 
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        evals_result = {}  # type: Dict[Any, Any]
+        tuner = optimize.LightGBMTuner(
+            params,
+            train_set,
+            num_boost_round=self.n_estimators,
+            valid_sets=valid_sets,
+            valid_names=eval_names,
+            fobj=fobj,
+            feval=feval,
+            # TODO(Kon): Pass init_model to LightGBMTuner
+            # init_model=init_model,
+            early_stopping_rounds=early_stopping_rounds,
+            evals_result=evals_result,
+            verbose_eval=verbose,
+            callbacks=callbacks,
+            time_budget=self.timeout,
+            study=self.study,
+            model_dir=str(self.train_dir),
+        )
+
+        tuner.run()
 
-        warnings.warn(
-            "LightGBMTuner doesn't support sklearn API. "
-            "Use `train()` or `LightGBMTuner` for hyperparameter tuning."
+        self._Booster = tuner.get_best_booster()
+        self._best_iteration = (
+            None if early_stopping_rounds is None else self._Booster.best_iteration
         )
-        super(LGBMClassifier, self).__init__(*args, **kwargs)
+        self._best_score = self._Booster.best_score
+        self._evals_result = evals_result if evals_result else None
+        self._objective = params["objective"]
+
+        self._Booster.free_dataset()
+
+        return self
+
+
+class LGBMClassifier(LGBMModel, ClassifierMixin):
```

Comment:
Could you add this class to `docs/source/reference/integration.rst` to create the reference?

Reply:
Fixed in 41e76fb.

Reply:
https://github.com/optuna/optuna/pull/1018/checks?check_run_id=1230177914
Sorry, I don't know how to fix the CI error.

Reply:
I'm sorry, but the log has been deleted. Could you rebase the branch and push it to kick the CI?

Reply:
https://github.com/optuna/optuna/pull/1018/checks?check_run_id=1790196768
Kicked the CI again. Please check.

Reply:
According to the log of `doctest`, something is wrong with the code example of `LGBMClassifier`:

```console
**********************************************************************
File "../../../../../../../opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/optuna/integration/_lightgbm_tuner/sklearn.py", line ?, in default
Failed example:
    clf.fit(X, y, eval_set=[(X_valid, y_valid)])
Expected:
    LGBMClassifier(...)
Got:

```
https://github.com/optuna/optuna/pull/1018/checks?check_run_id=1790196559

Do you have any idea about it?